channels_redis/core.py

```diff
@@ -616,6 +651,7 @@ async def group_discard(self, group, channel):
         key = self._group_key(group)
         async with self.connection(self.consistent_hash(group)) as connection:
             await connection.zrem(key, channel)
+        self.receive_buffer.expire()
```

Comment:
Maybe we can't rely on this *only* happening on `group_discard` (since some may not be using the group extension?).

Doing this sort of thing once per `receive()` call seems too expensive, though.